### PR TITLE
Adding determination of event thrust not by minimization

### PR DESCRIPTION
**Pull request by kjvbrt**
Adding function to calculate thrust of the event which does not require minimization.

### CODE REVIEW

**Comment by kjvbrt** [comment]
Removed unnecessary bits. The pull request contains only implementation of the functor which adds another possibility to determine thrust axis in the event. The thrust axis is not determined by minimization but by trying out all the possible combinations.

File: analyzers/dataframe/include/FCCAnalyses/Algorithms.h [hunk 1]
```diff
@@ -92,6 +92,22 @@ namespace Algorithms{
...
+    calculate_thrust(){}
+
+    inline void mag2(float (&vec)[4]);
+    inline float dot(float vec1[4], float vec2[4]);
+    inline void cross(float (&vec)[4], float vec1[4], float vec2[4]);
+    inline void unit(float (&vec)[4]);
+    inline void plus(float (&vec)[4], float vecIn1[4], float vecIn2[4]);
+    inline void minus(float (&vecOut)[4], float vecIn1[4], float vecIn2[4]);
+    inline void copy(float (&vecOut)[4], float vecIn[4]);
+
```
Comment (clementhelsens) [inline analyzers/dataframe/include/FCCAnalyses/Algorithms.h:106]: I'm wondering if we should move those type of utilities to some Utils namespace as they have nothing specific to thrust calculation

File: analyzers/dataframe/src/Algorithms.cc [hunk 2]
```diff
@@ -163,6 +163,148 @@ ROOT::VecOps::RVec<float> minimize_thrust::operator()(const ROOT::VecOps::RVec<f
...
+
+  // std::cout << "calculate_thrust: Calculating sum of all particles in event"
+  //           << std::endl;
+
+  float pVec[nParticles][4];   // 0 -- magnitude squared, 1 -- x, 2 -- y, 3 -- z
+  float pSum = 0.;
+  for (size_t i = 0; i < nParticles; ++i) {
+    pVec[i][1] = px.at(i);
+    pVec[i][2] = py.at(i);
+    pVec[i][3] = pz.at(i);
```
Comment (clementhelsens) [inline analyzers/dataframe/src/Algorithms.cc:242]: Better to replace .at by []

**Review by clementhelsens — CHANGES_REQUESTED** [review]
Thanks @kjvbrt , the implementation looks good to me, but I think we should probably move the inline utilities to a dedicated namespace

File: analyzers/dataframe/include/FCCAnalyses/Algorithms.h (hunk 1 above)
Comment (kjvbrt) [reply analyzers/dataframe/include/FCCAnalyses/Algorithms.h]: Moved those functions to `Utils.h`

File: analyzers/dataframe/src/Algorithms.cc (hunk 2 above)
Comment (kjvbrt) [reply analyzers/dataframe/src/Algorithms.cc]: Done

**Review by vvolkl — APPROVED** [review]

**Review by clementhelsens — APPROVED** [review]

**Pull request merged**